openssl/src/ssl/mod.rs

```diff
@@ -3484,6 +3486,48 @@ impl SslRef {
             }
         }
     }
+
+    /// Returns the NID of the negotiated group used for the handshake key
+    /// exchange process.
+    /// For TLSv1.3 connections this typically reflects the state of the
+    /// current connection, though in the case of PSK-only resumption, the
+    /// returned value will be from a previous connection.
+    /// For earlier TLS versions, when a session has been resumed, it always
+    /// reflects the group used for key exchange during the initial handshake
+    /// (otherwise it is from the current, non-resumption, connection).
+    /// This can be called by either client or server.
+    ///
+    /// If the NID for the shared group is unknown then the value is set to the
+    /// bitwise OR of TLSEXT_nid_unknown (0x1000000) and the id of the group.
+    #[corresponds(SSL_get_negotiated_group)]
+    #[cfg(ossl300)]
+    pub fn negotiated_group(&self) -> Result<c_int, ErrorStack> {
```

Comment:
I don't think we shouldn't be exposing this as a raw c_int. Since their doing the weird unknown thing, I think we'd need to make a `NegotiatedGroup` type that provides a `pub fn nid(self) -> Option<Nid>` and `pub fn unknown_group_id(self) -> Option<i32>`.

Reply:
I can do this change.

Then again in `SSL::group_to_name` which calls `SSL_group_to_name()`, we need the raw unaltered `c_int`.

Should the `NegotiatedGroup` type implement `From/To` trait to convert back to the raw `c_int`?
Suggestions on how to implement that?

Reply:
Yep, these wrapper types have an `as_raw` method to get the c_int out: https://docs.rs/openssl/latest/openssl/nid/struct.Nid.html#method.as_raw

Reply:
I tried addressing this with 893ce6a796a0c16b05578b996289e49d8f008f6d